src/main/java/net/neoforged/neoforge/network/filters/VanillaPacketSplitter.java

```diff
@@ -125,34 +122,26 @@ private static void onClientPacket(NetworkEvent.ServerCustomPayloadEvent event)
         if (state == STATE_LAST) {
             FriendlyByteBuf full = new FriendlyByteBuf(Unpooled.wrappedBuffer(receivedBuffers.toArray(new FriendlyByteBuf[0])));
             int packetId = full.readVarInt();
-            Packet<?> packet = protocol.codec(direction).createPacket(packetId, full);
+            Packet<?> packet = protocol.codec(flow).createPacket(packetId, full, channelHandlerContext);
             if (packet == null) {
                 LOGGER.error("Received invalid packet ID {} in neoforge:split", packetId);
             } else {
                 receivedBuffers.clear();
                 full.release();
-                ctx.enqueueWork(() -> genericsFtw(packet, event.getSource().getNetworkManager().getPacketListener()));
+
+                context.workHandler()
+                        .submitAsync(() -> context.packetHandler().handle(packet));
```

Comment:
Do we want this to be execute or add some other error message rather than this kind of just swallowing it?